### ISSUE TITLE
Generalise country specific code using abstract base class

### ISSUE THREAD

**Issue by andrewphilipsmith** [Urban-Analytics-Technology-Platform/poppusher#84]
Following a discussion with @yongrenjie @sgreenbury and @andrewphilipsmith. 

We came up with this approximate pattern as a way to generalise some of the country-specific code:

```
class generic_country(ABC):

    def get_source_table(self):
        @asset()
        def inner():
            return self._get_source_table()

        return inner
    
    # Must be implemented by the child class
    def _get_source_table(self):
        pass

class Redland(generic_country):

    def _get_source_table(self):
        return "Redland"

if __name__ == "__main__":
    redland = Redland()
    
    defs = Definitions(
        assets=[redland.get_source_table()]
    )
```

**Comment by andrewphilipsmith** [comment]
@sgreenbury will attempt this with Northern Ireland (#26).

**Comment by sgreenbury** [comment]
@andrewphilipsmith: Initial version of this now implemented in #98 and #117
